templates/deb/prerm_upgrade.sh.erb

```diff
@@ -8,10 +8,12 @@ before_remove() {
 <%=  script(:before_remove) %>
 <% end -%>
 
-<%# Removing the systemd service-%>
-<% if attributes[:deb_systemd] -%>
-systemctl stop <%= attributes[:deb_systemd] %> >/dev/null || true
-systemctl disable <%= attributes[:deb_systemd] %> >/dev/null || true
+<%# Stop and remove any systemd services that were installed-%>
+<% unless attributes[:deb_systemd].empty? -%>
+<% attributes[:deb_systemd].each do |service| -%>
```

Comment:
Did you want to use the same "try deb-systemd-invoke or systemctl" check here as your other scripts do?